flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/runtime/stream/sql/JoinITCase.scala

```diff
@@ -1106,4 +1108,23 @@ class JoinITCase(state: StateBackendMode) extends StreamingWithStateTestBase(sta
     val expected = List("500")
     assertEquals(expected.sorted, sink.getRetractResults.sorted)
   }
+
+  @Test
+  def testJoinWithUDFFilter(): Unit = {
+    val ds1 = failingDataSource(TestData.smallTupleData3).toTable(tEnv, 'a, 'b, 'c)
+    val ds2 = failingDataSource(TestData.tupleData5).toTable(tEnv, 'd, 'e, 'f, 'g, 'h)
+
+    tEnv.registerTable("T3", ds1)
+    tEnv.registerTable("T5", ds2)
+    tEnv.registerFunction("funcWithOpen", new FuncWithOpen)
+
+    val sql = "SELECT c, g FROM T3 join T5 on funcWithOpen(a + d) where b = e"
```

Comment:
cc @godfreyhe , why `funcWithOpen` is not handled with an extra `project`?

Reply:
@KurtYoung , `a` is from join left side (`T3`), while `d` is from join right side (`T5`).  `funcWithOpen(a + d)` could not be pushed into left side or right side as a `Project`.